python/tests/unit/test_serialization.py

```diff
@@ -270,16 +271,15 @@ def test_usage_as_required_field(
         class TestModel(SKBaseModel):
             """A test model."""
 
-            field: sk_type = pdt.Field(default_factory=lambda: sk_factory(sk_type))
+            field: sk_type = Field(default_factory=lambda: sk_factory(sk_type))
 
         assert_serializable(TestModel(), TestModel)
         assert_serializable(TestModel(field=sk_factory(sk_type)), TestModel)
 
 
 def assert_serializable(obj: _Serializable, obj_type) -> None:
-    """Assert that an object is serializable."""
+    """Assert that an object is serializable, uses both dump and dump_json methods."""
     assert obj is not None
-    serialized = obj.json()
+    serialized = obj.model_dump_json()
     assert isinstance(serialized, str)
-    deserialized = obj_type.parse_raw(serialized)
-    assert deserialized == obj
```

Comment:
Any reason why we don't do this anymore?

Reply:
Was replaced by a slightly different approach to comparing, see the note on __eq__ method here: https://docs.pydantic.dev/latest/migration/, so we are now checking equality by looking at the model_dumps (dict) of the original and one created based on json dump.